pennylane/io.py

```diff
@@ -410,113 +407,131 @@ def from_qiskit_op(qiskit_op, params=None, wires=None):
         raise RuntimeError(_MISSING_QISKIT_PLUGIN_MESSAGE) from e
 
 
-def from_qasm(quantum_circuit: str, measurements=False):
-    """Loads quantum circuits from a QASM string using the converter in the
+def from_qasm(quantum_circuit: str, measurements=None):
+    """
+    Loads quantum circuits from a QASM string using the converter in the
     PennyLane-Qiskit plugin.
 
     Args:
         quantum_circuit (str): a QASM string containing a valid quantum circuit
         measurements (None | MeasurementProcess | list[MeasurementProcess]): an optional PennyLane
-            measurement or list of PennyLane measurements that overrides any terminal measurements
-            that may be present in the input circuit. If set to ``None``, existing measurements
-            in the input circuit will be used.
+            measurement or list of PennyLane measurements that overrides the terminal measurements
+            that may be present in the input circuit. Defaults to ``None``, such that existing measurements
+            in the input circuit are kept. Mid-circuit measurements are kept either way.
```

Comment:
On reflection, I'm feeling like we might be overexplaining this concept, which may confuse users. Perhaps here we just keep it simple and don't mention mid circuit measurements, and instead use the examples to explain the subtlety.
```suggestion
            in the input circuit are returned.
```

Reply:
Do we add a note saying something like "see usage details"? - which btw I think should be added because right now we only have *Example* and it includes all the possible explanations

Reply:
I think a usage details section is a great idea - potentially separate sections on how mid-circuit measurements work, and about overriding terminal measurements. Keeps the initial example simple, and keeps the `[]` to access legacy behaviour separate from the general description of the function :)

Reply:
Sounds great, lmk what you think about the new structure!